match/spec/storage/git_storage_spec.rb

```diff
@@ -43,6 +43,46 @@
         expect(File.exist?(File.join(storage.working_directory, 'README.md'))).to eq(false) # because the README is being added when committing the changes
       end
 
+      it "directly clones a single branch wtih shallow clone" do
+        # GIVEN
```

Comment:
what about being a bit more DRY and move those initializations in one place, either using a `before` or an `around` under the same `describe` ? I mention [around](https://github.com/fastlane/fastlane/blob/master/fastlane/spec/cli_tools_distributor_spec.rb#L4-L10) in case the garbage collector kicks in. 

```
describe "#download" do
  before(:each) do
    @path = Dir.mktmpdir # to have access to the actual path
    allow(Dir).to receive(:mktmpdir).and_return(@path)
  end
```

Reply:
Refactored tests in 85c8aceea91f83bbe3f42d5eb955522307fc526c.
Combined your and @rogerluan comments.

Git spec helper:
1. branch checkout commands were repeated all over the spec.
2. Command execution check is too long, imo. Added a handful wrapper that accepts a single command string or array of commands.

Spec:
1. Default git url and branch.
2. Default temp dir stub.
3. Shorter test cases.